image_handler.py

```
@@ -0,0 +1,328 @@
+                    
+                    # Check file size (max 5MB for safety)
+                    content_length = response.headers.get('Content-Length')
+                    if content_length and int(content_length) > 5 * 1024 * 1024:
+                        logger.warning(f"Image too large: {url} ({content_length} bytes)")
+                        return None
+                    
+                    return await response.read()
+                else:
+                    logger.warning(f"Failed to download image: {url} (status: {response.status})")
+                    return None
+    except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)